Large scale: add paragraph about AES Key rotation (#107)
uyuni-project/uyuni-docs: modules/salt/pages/large-scale.adoc

You can also consider tuning the Server to consume less memory.
For more information on tuning, see xref:salt:large-scale-tuning.adoc[].

> === AES Key rotation

Reviewer: ```suggestion
=== AES Key Rotation
```

> Communication from the Salt Master to all Salt minions is encrypted with a single AES key. This key is rotated whenever:

Reviewer: ```suggestion
Communications between the Salt Master and clients is encrypted with a single AES key. 
The key is rotated when:
```

Author: Hmm, not really, in the sense communication is not bidirectionally encrypted in the same way. Here I am really referring to messages from master to minions only.

Reviewer: Ah, my bad. What about this:

```suggestion
Communications from the Salt Master to clients is encrypted with a single AES key.
The key is rotated when:
```

 - the salt-master process is restarted, or

>  - any Salt minion key is deleted (eg. a client is deleted from {productname})

Reviewer: ```suggestion
 * The ``salt-master`` process is restarted, or
 * Any client key is deleted (for example, when a client is deleted from {productname})
```

Author: I am taking your suggested edit but will stick to "minion key" here, as there are multiple keys involved for any client and here we refer to a very specific one (sorry, crypto is picky!)

> Upon rotation, all minions will have to re-authenticate to the Master next time they receive a message, and if the number of minions is high (several thousands) then this is expected to cause noticeable CPU load on the {productname} Server.

Reviewer: ```suggestion
After the AES key has been rotated, all clients must re-authenticate to the master.
By default, this happens next time a client receives a message.
If you have a large number of clients (several thousands), this can cause a high CPU load on the {productname} Server.
```

> 

Reviewer: ```suggestion
```

> If the load caused by massive re-authentication is excessive, it is recommended to delete keys in batches, possibly in off-peak hours, not to overload the {productname} Server.

Reviewer: ```suggestion
If the CPU load is excessive, we recommend that you delete keys in batches, and in off-peak hours if possible, to avoid overloading the server.
```

> More information can be found in Salt upstream documentation: https://docs.saltstack.com/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing https://docs.saltstack.com/en/getstarted/system/communication.html

Reviewer: ```suggestion
For more information, see: 
* https://docs.saltstack.com/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing 
* https://docs.saltstack.com/en/getstarted/system/communication.html
```
